### DISCUSSION TITLE
provisioner deployments not being created

### DISCUSSION THREAD

**Discussion by sjholden** [rook/rook#8293] [Q&A]
I have the same setup in two clusters - the same helm chart, the same values.yaml aside from ingress names on the dashboard.

One of them is working fine, the other is not creating the csi-cephfsplugin-provisioner or csi-rbdplugin-provisioner deployments at all.

```
; kubectl -n rook-ceph get  deployment
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
rook-ceph-crashcollector-a5-mc2-21205   1/1     1            1           8d
rook-ceph-crashcollector-a6-mc1-21206   1/1     1            1           8d
rook-ceph-crashcollector-a6-mc2-21205   1/1     1            1           8d
rook-ceph-mds-ceph-cluster-cephfs-a     1/1     1            1           26m
rook-ceph-mds-ceph-cluster-cephfs-b     1/1     1            1           26m
rook-ceph-mgr-a                         1/1     1            1           8d
rook-ceph-mon-a                         1/1     1            1           8d
rook-ceph-mon-b                         1/1     1            1           8d
rook-ceph-mon-c                         1/1     1            1           8d
rook-ceph-operator                      1/1     1            1           8d
rook-ceph-osd-0                         1/1     1            1           8d
rook-ceph-osd-1                         1/1     1            1           8d
rook-ceph-osd-2                         1/1     1            1           8d
rook-ceph-tools                         1/1     1            1           8d

```

(I tried recreating the cephfilesystem hence the lower AGE stamps on those deployments before I noticed the csi-rbdplugin-provisioner also didn't exist).

I don't see anything obvious in the logs, And the troubleshooting docs involve the logs from the deployments that don't exist in this case.

So has anyone seen this before? Have an idea of what to check?

**Comment by Madhu-1** [comment]
Check rook operator logs.

**Reply by sjholden** [reply]
I don't see anything that seems relevant, the logs from the day I deleted and recreated the CephFilesystem object.
[logs.txt](https://github.com/rook/rook/files/6796432/logs.txt)

**Reply by Madhu-1** [reply]
Can you restart the operator pod and see that helps? If not please provide full rook operator pod log

**Reply by sjholden** [reply]
Restarting the operator pod seems to have got it working. I take it that should be the first thing I try if something just isn't happening?

**Reply by travisn** [reply]
@sjholden Did you change the settings for starting the cephfs csi driver, or it just never started? If settings are changed and you don't see things updating, then restarting the operator can certainly help sometimes.

**Reply by sjholden** [reply]
I didn't change settings. I applied the rook-ceph helm chart and then after argocd said that had was healthy applied a helm chart that just creates the CephCluster, CephFilesystem, and StorageClass objects. Which is what I did in the other cluster which "just worked".

**Reply by travisn** [reply]
Ok, it's expected that the csi provisioner and plugin pods would be started immediately when the cephcluster CR is created. It's unexpected that you'd have to restart the operator pod to get the driver to start. If you hit this again, please do share the operator log before you restart it.
